templates/includes/backendselection4.vcl.erb

```diff
@@ -2,7 +2,13 @@ if (<%= @condition -%>) {
     <%- if @movedto -%>
   synth( 750, "<%= @movedto -%>" + req.url);
     <%- else -%>
-  set req.backend_hint = <%= @_backend -%>;
+      <%- if @backend -%>
+  set req.backend_hint = <%= @backend %>;
+  set req.http.x-backend = "<%= @backend %>";
```

Comment:
No, I don't think that we can do that. `.backend()` returns a different backend every time that it is called.